### PR TITLE
Adding test coverage to "dd" and "mm" edge cases

### PR DESCRIPTION
**Pull request by svetakvsundhar**
Adding test coverage to "dd" edge case in GCS bucket name.

### CODE REVIEW

File: v2/common/src/test/java/com/google/cloud/teleport/v2/io/WindowedFilenamePolicyTest.java [hunk 1]
```diff
@@ -131,21 +131,33 @@ public void testWithDynamicDirectory() throws IOException {
...
+    
+    // Directory with "mm" specified
     WindowedFilenamePolicy policy =
         WindowedFilenamePolicy.writeWindowedFiles()
             .withOutputDirectory("gs://test-bucket-mm/YYYY/MM/dd/HH:mm")
             .withOutputFilenamePrefix("output")
             .withShardTemplate("-SSS-of-NNN");
+    
+    // Directory with "dd" specified
+    WindowedFilenamePolicy policy_with_dd =
```
Comment (bvolpato) [inline v2/common/src/test/java/com/google/cloud/teleport/v2/io/WindowedFilenamePolicyTest.java:143]: nitpick: Java / Google style-guide often favors camel case `policyWithDay`

File: v2/common/src/test/java/com/google/cloud/teleport/v2/io/WindowedFilenamePolicyTest.java [hunk 2]
```diff
@@ -131,21 +131,33 @@ public void testWithDynamicDirectory() throws IOException {
...
+        WindowedFilenamePolicy.writeWindowedFiles()
+            .withOutputDirectory("gs://test-bucket-dd/YYYY/MM/dd/HH:mm")
```
Comment (bvolpato) [inline v2/common/src/test/java/com/google/cloud/teleport/v2/io/WindowedFilenamePolicyTest.java:145]: This is a good test. Maybe we could really take this to the next level by having all fields if we want to extrapolate 

e.g.
`.withOutputDirectory("gs://test-bucket-YYYY-MM-dd_HH_mm/YYYY/MM/dd/HH:mm")`

we expect the bucket to stay intact

Comment (svetakvsundhar) [reply v2/common/src/test/java/com/google/cloud/teleport/v2/io/WindowedFilenamePolicyTest.java]: good idea. in that case i'll just modify the original test to include this (see latest commit)

**Review by bvolpato — APPROVED** [review]
